### PR TITLE
parallel-benchmark: Store data in SQLite for long runs

### PR DESCRIPTION
**Pull request by def-**
Seems not to cause a performance difference, but lacks some features currently

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] If this PR includes major [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note), I have pinged the relevant PM to schedule a changelog post.

### CODE REVIEW

File: test/parallel-benchmark/mzcompose.py [hunk 1]
```diff
@@ -105,24 +107,70 @@ def known_regression(scenario: str, other_tag: str) -> bool:
...
+            self.max: float = cursor.fetchall()[0][0]
+            cursor.execute(
+                "SELECT min(duration) * 1000 FROM measurements WHERE scenario = ? AND action = ?",
+                (m.scenario, action),
+            )
+            self.min: float = cursor.fetchall()[0][0]
+            cursor.execute(
+                "SELECT avg(duration) * 1000 FROM measurements WHERE scenario = ? AND action = ?",
+                (m.scenario, action),
+            )
```
Comment (nrainer-materialize) [inline test/parallel-benchmark/mzcompose.py:155]: Do this in one query?

**Review by nrainer-materialize — APPROVED** [review]
What's the motivation for this change?

**Comment by def-** [comment]
The QA Canary job runs parallel-benchmark for 24 hours, it would require a beefy machine to keep all the results in memory, so I'd rather have a small machine and store the results on disk.

**Comment by nrainer-materialize** [comment]
> The QA Canary job runs parallel-benchmark for 24 hours, it would require a beefy machine to keep all the results in memory, so I'd rather have a small machine and store the results on disk.

Makes sense! I was not aware that we are collecting that much data.

File: test/parallel-benchmark/mzcompose.py [hunk 2]
```diff
@@ -105,24 +107,50 @@ def known_regression(scenario: str, other_tag: str) -> bool:
...
+            self.std: float = float(numpy.std(durations, ddof=1))
+            self.slope: float = float(numpy.polyfit(times, durations, 1)[0])
+        elif isinstance(m, SQLiteStore):
+            cursor = m.conn.cursor()
+            cursor.execute(
+                f"SELECT count(*), count(*) / (max(timestamp) - {start_time}), max(duration) * 1000, min(duration) * 1000, avg(duration) * 1000 FROM measurements WHERE scenario = ? AND action = ?",
+                (m.scenario, action),
+            )
+            self.queries, self.qps, self.max, self.min, self.avg = cursor.fetchone()
+            # TODO: Rest
```
Comment (nrainer-materialize) [inline test/parallel-benchmark/mzcompose.py:137]: There is an open TODO. Do you plan to address this within this PR?

Note that I cancelled auto-merge. Feel free to proceed, though.

Comment (def-) [reply test/parallel-benchmark/mzcompose.py:137]: Nope, I left that for future me.

**Pull request merged**